src/View/mainpage/DrawROIWindow/UIDrawROIWindow.py

```diff
@@ -862,30 +881,147 @@ def onFillClicked(self):
+                id = self.current_slice
+
+                # This needs to be updated when merged with 2d
+                print("is_3d: " + str(is_3d))
+                if is_3d:
+                    print(str(is_3d) + ": is_3d")
+                    if hasattr(self, 'seed'):
+                        delattr(self, 'seed')
+                    self.create_drawing_3D(min_pixel, max_pixel, pixmaps, id)
+                else:
+                    self.create_drawing(min_pixel, max_pixel, pixmaps, id)
+
             else:
                 QMessageBox.about(self.draw_roi_window_instance,
                                   "Not Enough Data",
                                   "Not all values are specified or correct.")
+
+    def create_drawing(self, min_pixel, max_pixel, pixmaps, id):
+        dt = self.patient_dict_container.dataset[id]
+        dt.convert_pixel_data()
+        # Path to the selected .dcm file
+        location = self.patient_dict_container.filepaths[id]
+        self.ds = pydicom.dcmread(location)
+
+        self.drawingROI = Drawing(
+            pixmaps[id],
+            dt._pixel_array.transpose(),
+            min_pixel,
+            max_pixel,
+            self.patient_dict_container.dataset[id],
+            self.draw_roi_window_instance,
+            self.slice_changed,
+            self.current_slice,
+            self.drawing_tool_radius,
+            self.keep_empty_pixel,
+            self.pixel_transparency,
+            set()
+        )
+
+        self.slice_changed = True
+        self.has_drawing = True
+        self.dicom_view.view.setScene(self.drawingROI)
+        self.enable_cursor_diameter_change_box()
+
+    def create_drawing_3D(self, min_pixel, max_pixel, pixmaps, id):
+        """
+        Creates drawing allowing for the user to start drawing on dicom view.
+        """
+        # If the seed is set then start searching, else assign the drawing function to the left click
+        if hasattr(self, 'seed'):
+            for x in range(0, 2):
+                print("x == " + str(x))
+                if x == 0:
+                    range_start = id - 1
+                    range_end = 1
+                    step = -1
+                else:
+                    range_start = id
+                    range_end = len(self.patient_dict_container.dataset) - 1
+                    step = 1
+
+                for y_search in range(range_start, range_end, step):
+                    print(str(range_start) + " range end: " + str(range_end) + " range: " + str(step))
+                    temp_id = y_search
+
+                    self.dicom_view.slider.setValue(temp_id)
+
+                    dt = self.patient_dict_container.dataset[temp_id]
+                    dt.convert_pixel_data()
+
+                    # Path to the selected .dcm file
+                    location = self.patient_dict_container.filepaths[temp_id]
+                    self.ds = pydicom.dcmread(location)
```

Comment:
what are you doing here?

Reply:
this whole method is confusing, consider splitting up.
Use your previously defined create draw method?

Reply:
Added comments, cant use the previously defined create draw method (or I don't know how I could use it) since a couple differences makes the code not re-useable

Reply:
There is stuff that is repeated and can be cleaned up.
- Seperate out the functionality and pass in an id
- set your flags to true outside of the if so that are not repeated
- etc

Can jump on a call if needed or if I'm missing somthing